File: examples/python/register_instance.py
"""Example: Register an OpenClaw Gateway instance."""
from minions_openclaw import InstanceManager


def main():
    manager = InstanceManager()

    instance = manager.register(
        'my-gateway',
        'ws://localhost:8080',
        'optional-auth-token'
    )

    print(f'Registered instance: {instance.id}')
    print(f'Title: {instance.title}')
    print(f"URL: {instance.fields['url']}")
    print(f"Status: {instance.fields['status']}")

    instances = manager.list()
    print(f'\nAll instances: {len(instances)}')


if __name__ == '__main__':
    main()

File: examples/typescript/register-instance.ts
/**
 * Example: Register an OpenClaw Gateway instance
 */
import { InstanceManager } from '@minions-openclaw/sdk';

async function main() {
  const manager = new InstanceManager();

  const instance = await manager.register(
    'my-gateway',
    'ws://localhost:8080',
    'optional-auth-token'
  );

  console.log('Registered instance:', instance.id);
  console.log('Title:', instance.title);
  console.log('URL:', instance.fields['url']);
  console.log('Status:', instance.fields['status']);

  const instances = await manager.list();
  console.log('\nAll instances:', instances.length);
}

main().catch(console.error);
